datadotworld/client/api.py

```diff
@@ -335,6 +337,44 @@ def upload_files(self, dataset_key, files):
         except _swagger.rest.ApiException as e:
             raise RestApiError(cause=e)
 
+    def open_file_writer(self, dataset_key, file_name):
+        """Open a streaming writer to a data.world file
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        file_name: str
+            The name of the file to write
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>>
+        >>> with api_client.open_file_writer('username/test-dataset',
+        ...                                  'test.txt') as w:
+        ...   w.write("this is a test.")
+        >>>
+        >>> with api_client.open_file_writer('username/test-dataset',
+        ...                                  'test.jsonl') as w:
+        ...   w.write({'foo':42, 'bar':"A"})
+        ...   w.write({'foo':13, 'bar':"B"})
+        >>>
+        >>> import csv
+        >>> with api_client.open_file_writer('username/test-dataset',
+        ...                                  'test.csv') as w:
+        ...   csvw = csv.DictWriter(w, fieldnames=['foo', 'bar'])
+        ...   csvw.writeheader()
+        ...   csvw.writerow({'foo':42, 'bar':"A"})
+        ...   csvw.writerow({'foo':13, 'bar':"B"})
+        """
+        try:
+            return DataDotWorldFileWriter(self._config, dataset_key, file_name,
```

Comment:
It would be helpful if the name indicated that this is writing remotely (as opposed to locally). Also, with the exception of our main class `DataDotWorld`, I would rather not use that as a prefix for other classes.